components/feature/session/src/main/java/mozilla/components/feature/session/SessionProvider.kt

```diff
@@ -18,10 +17,9 @@ import java.util.concurrent.TimeUnit
  * Manages access to active browser / engine sessions and their persistent state.
  */
 class SessionProvider(
-    private val context: Context,
     initialSession: Session = Session(""),
-    private val sessionStorage: SessionStorage = DefaultSessionStorage(context),
-    private val savePeriodically: Boolean = false,
+    private val sessionStorage: SessionStorage? = null,
+    private val savePeriodically: Boolean = true,
```

Comment:
I think `true` is a better default here. Likely, we want this to be on, but we can change that later too.